docs/wiki/ModulesDesign.md

```diff
@@ -124,33 +132,262 @@ name: '${deployment().name}-Table-${index}'
+    metrics: diagnosticsMetrics
+    logs: diagnosticsLogs
+  }
+  scope: <mainResource>
+}
+```
 
-   Resources like `Microsoft.Sql/servers` may have dedicated templates for child-resources such as `Microsoft.Sql/servers/databases`. In these cases we recommend to create a sub-folder called after the child-resource name, so that the path to the child-resource folder is consistent with its resource type. In the given example we would have a sub-folder `databases` in the parent-folder `servers`.
+### Private Endpoints
 
-   ```
-   Microsoft.Sql
-   └─ servers [module]
-      └─ databases [child-module/resource]
-   ```
+The Private Endpoint deployment has 2 elements to it. A module that contains the implementation, and a module reference in the parent resource. The first loops through the endpoints we want to create, the second processes them.
 
-   In this folder we recommend to place the child-resource-template alongside a ReadMe (that can be generated via the `.github\workflows\scripts\Set-ModuleReadMe.ps1` script) and optionally further nest additional folders for it's child-resources. The parent template should reference all it's child-templates to allow for an end to end deployment experience while allowing any user to also reference 'just' the child-resource itself. In the case of the SQL-server example the server template would reference the database module and encapsulate it it in a loop to allow for the deployment of n-amount of databases.
+***1st element in main resource***
 
-<!--
-- **Overlapping/Ambigious providers**<p>
-  There may be cases where a folder is already leveraged by a different module with the same resource provider. In these cases we recommend to add an additional layer into the hierarchy by moving the module that originally populated the conflicting folder into a sub-folder of the same using a meaningful name. The new module can then be positioned on the same level, again with a meaningful name. For example:
-  ```
-  Microsoft.<provider>
-  ├─ <service1> [module]
-  └─ <service2> [the conflicting name - now shared as a parent folder]
-     ├─ <service2Variant1> [module]
-     └─ <service2Variant2> [module]
-  ```
+```bicep
+@description('Optional. Configuration Details for private endpoints.')
+param privateEndpoints array = []
 
-  > ***Note:*** The intend should always be to add new logic to the original template instead of adding artificial new modules. Hence, this solution should only be applied if no other solutions work.
--->
+module <mainResource>_privateEndpoints '.bicep/nested_privateEndpoint.bicep' = [for (privateEndpoint, index) in privateEndpoints: {
+  name: '${uniqueString(deployment().name, location)}-PrivateEndpoint-${index}'
+  params: {
+    privateEndpointResourceId: <mainResource>.id
+    privateEndpointVnetLocation: empty(privateEndpoints) ? 'dummy' : reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
```

Comment:
Is this correct? It seems strange that we need empty(privateEndpoints) as it will not even do anything here if there are no privateEndpoints to process. OR am i missing something? Could it not just be `reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location`

Reply:
true story. We should really clean up the private Endpoints some time. Same for diagnosticSettings. The example here is cleaner than what we have in the modules

Reply:
done